torchdynamo/symbolic_convert.py

```diff
@@ -994,7 +1001,11 @@ def NOP(self, inst):
         pass
 
     def POP_TOP(self, inst):
-        self.pop()
+        val = self.pop()
+        if isinstance(val, GradModeVariable):
```

Comment:
I'd expect this to fail on:
```
tmp = torch.set_grad_enabled(True)
```

POP_TOP only happens if the user doesn't store the result.

Reply:
Should we then then effectively call `GradModeVariable` `enter` method on `CALL_METHOD` or `CALL_FUNCTION`?

Reply:
Why does it have to be from the bytecode of CALL_METHOD or CALL_FUNCTION? I haven't run it, but this should already be a `call_function` on a `TorchVariable` right? The problem is that it goes down into GradModeVariable's ctor, which is a no-op, and GradModeVariable itself is a ContextWrappingVariable, whereas this is not a context.

The right fix here would be to make a helper util like the one I described above, where we create:

```
 node = tx.output.graph.create_node(
            "call_function", torch._C._set_grad_enabled, (value,), {}
        )
```

And wrap it in a VariableTracker.

Reply:
Yes, that sounds good. I meant something along this lines as well. Basically put the grad state op in Fx Graph during the symbolic convesion of call_function (which is TorchVariable in this case), instead of delaying it to `SETUP_WITH` or `POP_TOP`.